Added Flyway and FootballerDto

SmokingSandwich opened the pull request:
1. Added a Service layer.
2. Added first Flyway migration.
3. Added FootballerDto class.
4. Made the Service and Controller use Dto to send only required information.
5. Changed Repository methods to use JPQL instead of native queries.
6. Fixed the controller.

forestg83 commented:
1. Nie dałeś opisu PR. Poza nazwą nie wiadomo co to ma robić. Zrób proszę tak jak w przykładzie, który Tobie pokazałem.

forestg83 commented on src/main/java/com/owczarczak/footballers/footballer/FootballerController.java:
```diff
@@ -14,62 +13,50 @@
-    FootballerRepository repository;
+    FootballerService service;
 
     @GetMapping
```
Ten kod jest do usunięcia. Nie trzymamy kodu testowego.

forestg83 commented on src/main/java/com/owczarczak/footballers/footballer/FootballerController.java:
```diff
@@ -14,62 +13,50 @@
-            return ResponseEntity.notFound().build();
-        }
-        return ResponseEntity.ok(repository.findById(id));
+    public Optional<FootballerDto> getFootballerById(@PathVariable int id) {
+        return service.getFootballerById(id);
     }
 
```
Nie zwracaj Optional w kontrolerze.

Jeśli Optional z serwisu jest pusty to zwróć ResponseEntity z kodem HTTP/404, jeśli coś jest to zwróć HTTP/200 z DTO.

SmokingSandwich replied:
Resolved

forestg83 commented on src/main/java/com/owczarczak/footballers/footballer/FootballerController.java:
```diff
@@ -14,62 +13,50 @@
-        footballerToUpdate.setId(id);
-        Footballer result = repository.save(footballerToUpdate);
-        return ResponseEntity.created(URI.create("/" + result.getId())).body(result);
+    Optional<FootballerDto> updateFootballer(@PathVariable int id, @RequestBody @Validated FootballerDto footballerToUpdate) {
```
Co miałoby znaczyć opcjonalne zwrócenie DTO przy aktualizacji?

Zwracaj ResponseEntity z odpowiednim kodem.

forestg83 commented on src/main/java/com/owczarczak/footballers/footballer/FootballerDto.java:
```diff
@@ -0,0 +1,30 @@
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    int id;
+    @NotBlank
```
Wyrzuć te walidacje.

forestg83 commented on src/main/java/com/owczarczak/footballers/footballer/FootballerRepository.java:
```diff
@@ -6,21 +6,13 @@
-//            where 'name' like '%search%'
-//            """, nativeQuery = true)
-//    List<Footballer> getFootballersByName(@Param("search") String name);
+    @Query("select * from footballers order by height desc limit 3")
```
Czy to na pewno działa?

SmokingSandwich replied:
it didn't work, but it's fixed now

forestg83 commented on src/main/java/com/owczarczak/footballers/footballer/FootballerService.java:
```diff
@@ -17,35 +17,35 @@ public class FootballerService {
 
-    public List<Footballer> getAllFootballers() {
+    public List<FootballerDto> getAllFootballers() {
         return repository.findAll();
```
Gdzie jest mapowanie z Footballer -> FootballerDTO?

SmokingSandwich replied:
ok, added.

forestg83 commented on src/main/java/com/owczarczak/footballers/footballer/FootballerRepository.java:
```diff
@@ -6,21 +6,13 @@
-//            """, nativeQuery = true)
-//    List<Footballer> getFootballersByName(@Param("search") String name);
+    @Query("select * from footballers order by height desc limit 3")
+    List<FootballerDto> get3HighestFootballers();
```
Repozytorium działa na encjach, nie na DTO!

SmokingSandwich replied:
ok, changed the repository to work on entities

forestg83 reviewed (changes requested):
Popraw wszystko co wskazałem do zmiany.

SmokingSandwich commented:
1) Added a Service layer.
2) Added first Flyway migration.
3) Added FootballerDto class.
4) Made the Service and Controller use Dto to send only required information.

forestg83 commented on pom.xml:
```diff
@@ -55,6 +55,17 @@
+			<artifactId>flyway-core</artifactId>
+		</dependency>
+
+		<dependency>
+			<groupId>org.mapstruct</groupId>
+			<artifactId>mapstruct</artifactId>
+			<version>1.5.3.Final</version>
```
No, ale jak to jest rozwiązane, jak nadal są zależności do mapstructa, którego nie używasz?
